### DISCUSSION TITLE
ALTER TABLE DROP COLUMN TEST failed because one or more objects access this column.

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#11093] [Archive]
i don't know why i get this error when i want to change in settings in Field of colection - Unique - Value has to be unique. Type of the field is string...

![Zrzut ekranu 2022-01-17 104823](https://user-images.githubusercontent.com/90528895/149747633-902b7497-d0b6-4a6c-9d84-9877d1a57381.jpg)

**Comment by joselcvarela** [comment]
Heya,
It seems there's some trigger, index or another DB mechanism which is relying on this column.
Can you confirm if that's the case?
If so, you would need to handle those manually, I believe.

**Reply by koksikus** [reply]
Hi,
I do not have any trigger. How to check if I have index or another DB mechanism? I don't think I have it either. 
I have a reference to this entire collection in other collections. I only display this number in a different collection. I want to keep displaying the number (or NULL value) but want to temporarily disable uniqueness checking.

**Reply by joselcvarela** [reply]
Doing a quick search I found this https://stackoverflow.com/questions/43549413/alter-table-drop-column-failed-because-one-or-more-objects-access-this-column
It seems you may need to remove the constraints linked to this column before doing that operation.
Use some SQL tool to inspect your DB like TablePlus or DBeaver.

**Reply by koksikus** [reply]
The new fields in the collection also do not want to change the setting in this regard. Even if I create a new collection with a new field in it, the same thing happens. I get an error and can't correct the uniqueness of the field.

If I create a new field without uniqueness and when I edit it again, I can check this field and set the uniqueness and it will be saved without error. If I enter the settings again and try to deprive my uniqueness, an error pops up!

**Reply by joselcvarela** [reply]
Trying to do the same with MS SQL didn't find any problems.
Can you provide a minimal dump to reproduce this?
It may be an issue on Directus or with your configuration, but I am not being able to reproduce this.

**Reply by koksikus** [reply]
What do you need and how to do it correctly? 
Where in the database should this setting appear
![Zrzut ekranu 2022-01-19 102926](https://user-images.githubusercontent.com/90528895/150103730-ce3c467b-bc8e-419c-89cc-5acee3a97fc4.jpg)

![Zrzut ekranu 2022-01-19 104332](https://user-images.githubusercontent.com/90528895/150105186-cdda2b39-c2fa-4a2a-affb-f6afe18e278f.jpg)

same for Nullable Allow null value

when i try to delete a column directly in the database i get this error
![Zrzut ekranu 2022-01-24 113938](https://user-images.githubusercontent.com/90528895/150768170-190a97d0-0d6d-4c86-8d9d-175707de9d67.jpg)

**Reply by koksikus** [reply]
@joselcvarela Maybe you have an idea how to solve it. Indexes seems to be influencing this. I think that's what Directus did. I don't know why it was created ...

![Zrzut ekranu 2022-01-26 114213](https://user-images.githubusercontent.com/90528895/151149176-c2b11b5c-75ef-4d0e-a7a5-64cf824f23e2.jpg)

**Reply by koksikus** [reply]
I found the reason, so this is because "indexes", if I delete it directly in the database, I can then delete the field in directus. If I set the uniqueness of the field, directus creates an "indexes" in the database and then again, I can't delete the field until I delete the "indexes" in the database
I'm using Directus 9.3.0
